### DISCUSSION TITLE
How to determine the final crystal structure after SSCHA minimisation?

### DISCUSSION THREAD

**Discussion by jgSi1113** [SSCHAcode/python-sscha#355] [Q&A] [answered]
Dear SSCHA developers,

Recently, I used the SSCHA code to minimize the dynamic matrix and to study the influence of qauntum effects on the system I concerned.  After SSCHA minimisation, the log says the system is under 2.8 GPa, and the structure I used is fully relaxed without any pressure.  Does it mean that the structure changed when the quantum effects are taken into account?  How can I output the structure after SSCHA minimisation? Or  is it just an auxiliary pressure value likes the SSCHA frequency?

![image](https://github.com/user-attachments/assets/8f150289-33a9-4c66-85eb-36432222715e)

Best & Regards!
Jianguo Si

**Comment by mesonepigreco** [comment, marked as answer]
Hi, 
it depends on the kind of run you made. 
The structure is stored inside the final dynamical matrix, which you can inspect. You can save it in other formats:

```python
# [here the sscha minimization code]
minim.run()

# save the structure of the final dynamical matrix in QuantumESPRESSO format
minim.dyn.structure.save_scf("my_final_structure.scf")
```

You can inspect the structure here. The final stress is the real value of the stress, accounting for quantum and thermal effects; see [PRB, 98, 024106, 2018](https://journals.aps.org/prb/abstract/10.1103/PhysRevB.98.024106).

In your case, you did not change the unit cell to relax the stress. If you want to, check how to do a SSCHA vc_relax from [this tutorial](https://sscha.eu/Tutorials/tutorial_01_first_simulations/).

**Reply by jgSi1113** [reply]
Thanks for your kindly reply.  It's more clear for me.

Regards!
Jianguo
